Source/WebCore/page/Navigation.cpp

```diff
@@ -405,6 +405,11 @@ Navigation::Result Navigation::navigate(const String& url, NavigateOptions&& opt
     auto newURL = window->protectedDocument()->completeURL(url, ScriptExecutionContext::ForceUTF8::Yes);
     const URL& currentURL = protectedScriptExecutionContext()->url();
 
+    // Guard against recursive navigate() calls
+    NavigateCallCounter counter(*this);
```

Comment:
Please minimize the scope of this counter to the recursive call is made. In this case, `loadFrameRequest` dispatches `NavigateEvent` in it. Use explicit scope to make it clear the relationship between the counter and the call.